code/game/gamemodes/cult/blood_magic.dm

```diff
@@ -0,0 +1,831 @@
+			"<span class='cultitalic'>You invoke the veiling spell, hiding nearby runes and cult structures.</span>")
+		charges--
+		SEND_SOUND(owner, sound('sound/magic/smoke.ogg', 0, 1, 25))
+		owner.whisper(invocation)
+		for(var/obj/O in range(4, owner))
+			O.cult_conceal()
+		revealing = TRUE
+		name = "Reveal Runes"
+		button_icon_state = "revealing"
+	else
+		owner.visible_message("<span class='warning'>A flash of light shines from [owner]'s hand!</span>", \
+			 "<span class='cultitalic'>You invoke the counterspell, revealing nearby runes and cult structures.</span>")
+		charges--
+		owner.whisper(invocation)
+		SEND_SOUND(owner, sound('sound/misc/enter_blood.ogg', 0, 1, 25))
+		for(var/obj/O in range(5, owner)) //slightly higher in case we arent in the exact same spot
+			O.cult_reveal()
+		revealing = FALSE
+		name = "Conceal Runes"
+		button_icon_state = "veiling"
+	if(charges <= 0)
+		qdel(src)
+	desc = base_desc
+	desc += "<br><b><u>Has [charges] use\s remaining</u></b>."
+	UpdateButtonIcon()
+
+/datum/action/innate/cult/blood_spell/manipulation
+	name = "Blood Rites"
+	desc = "Empowers your hand to manipulate blood. Use on blood or a noncultist to absorb blood to be used later, use on yourself or another cultist to heal them using absorbed blood. \
+		Use the spell in-hand to cast advanced rites, such as summoning a magical blood spear or firing blood projectiles out of your hands."
+	invocation = "Fel'th Dol Ab'orod!"
+	button_icon_state = "manip"
+	charges = 5
+	magic_path = "/obj/item/melee/blood_magic/manipulator"
+
+// The "magic hand" items
+/obj/item/melee/blood_magic
+	name = "\improper magical aura"
+	desc = "A sinister looking aura that distorts the flow of reality around it."
+	icon = 'icons/obj/items.dmi'
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	flags = ABSTRACT | DROPDEL
+
+	w_class = WEIGHT_CLASS_HUGE
+	throwforce = 0
+	throw_range = 0
+	throw_speed = 0
+	var/invocation
+	var/uses = 1
+	var/health_cost = 0 //The amount of health taken from the user when invoking the spell
+	var/datum/action/innate/cult/blood_spell/source
+
+/obj/item/melee/blood_magic/New(loc, spell)
+	source = spell
+	uses = source.charges
+	health_cost = source.health_cost
+	..()
+
+/obj/item/melee/blood_magic/Destroy()
+	if(!QDELETED(source))
+		if(uses <= 0)
+			source.hand_magic = null
+			qdel(source)
+			source = null
+		else
+			source.hand_magic = null
+			source.charges = uses
+			source.desc = source.base_desc
+			source.desc += "<br><b><u>Has [uses] use\s remaining</u></b>."
+			source.UpdateButtonIcon()
+	..()
+
+/obj/item/melee/blood_magic/attack_self(mob/living/user)
+	afterattack(user, user, TRUE)
+
+/obj/item/melee/blood_magic/attack(mob/living/M, mob/living/carbon/user)
+	if(!iscarbon(user) || !iscultist(user))
```

Comment:
Deconversions does remove blood magic but since the spell and the melee items are only loosely linked, this check is here. Also cover potential edge cases